pkg/controllers/openshift-route-controller-manager.go

```diff
@@ -88,15 +84,46 @@ func (s *OCPRouteControllerManager) writeConfig() *openshiftcontrolplanev1.OpenS
 		},
 	}
 
-	// https://github.com/openshift/route-controller-manager/blob/main/pkg/cmd/route-controller-manager/openshiftcontrolplane_default.go
-	configdefaults.SetRecommendedHTTPServingInfoDefaults(c.ServingInfo)
-	configdefaults.SetRecommendedKubeClientConfigDefaults(&c.KubeClientConfig)
-	c.LeaderElection = leaderelection.LeaderElectionDefaulting(c.LeaderElection, "kube-system", "openshift-route-controllers")
-	return c
+	scheme := runtime.NewScheme()
+	if err := openshiftcontrolplanev1.AddToScheme(scheme); err != nil {
+		return err
+	}
+	codec := serializer.NewCodecFactory(scheme).LegacyCodec(openshiftcontrolplanev1.GroupVersion)
+	encodedConfig, err := runtime.Encode(codec, rcmConfig)
+	if err != nil {
+		return err
+	}
+	ctrlConfig := &unstructuredv1.Unstructured{}
+	if err := runtime.DecodeInto(codec, encodedConfig, ctrlConfig); err != nil {
+		return err
+	}
+
+	const namespace = "openshift-route-controller-manager"
+	builder := controllercmd.NewController(s.Name(), route_controller_manager.RunRouteControllerManager).
+		WithKubeConfigFile(s.kubeconfig, nil).
+		WithComponentNamespace(namespace).
+		// Without an explicit owner reference, the builder will try using POD_NAME or the
+		// first pod in the target namespace (and fail because we have no pod).
+		WithComponentOwnerReference(&corev1.ObjectReference{
+			APIVersion: corev1.SchemeGroupVersion.String(),
+			Kind:       "Namespace",
+			Name:       namespace,
+			Namespace:  namespace,
+		}).
+		WithServer(*rcmConfig.ServingInfo, v1alpha1.DelegatedAuthentication{}, v1alpha1.DelegatedAuthorization{})
```

Comment:
not sure if microshift needs the server, but it is serving `metrics` and `healthz` endpoints

Reply:
I think we'd want healthz now, and at some point we'll want metrics, too. So I think it's OK to leave it in.

Can we control the port used? We're likely to have a couple of endpoints like this for embedded components, and I would either want to have them on separate ports or merge the results so we only need to expose one endpoint for metrics and one for health.

Reply:
ok, yeah the port can be changed in the config